File: js/medium/container-most-water.js
const maxArea = (height) => {
  let left = 0;
  let right = height.length - 1;
  let area = 0;
  while (left < right) {
    temp = Math.min(height[left], height[right]) * (right - left);
    area = Math.max(area, temp);

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }
  return area;
};

console.log(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]));

File: py/medium/container-most-water.py
from typing import List
def maxArea(height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        area = 0
        
        while left <= right:
            temp = min(height[left], height[right]) * (right - left)
            print(area, temp)
            if temp > area:
                area = temp
            
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
                
        return area

print(maxArea([1,8,6,2,5,4,8,3,7]))